app/src/pages/inside/dashboardItemPage/newWidgets/charts/launchesComparisonChart/launchesComparisonChart.jsx

```diff
@@ -0,0 +1,319 @@
+import * as d3 from 'd3-selection';
+import { C3Chart } from '../../c3chart';
+import { renderTooltip } from '../../tooltip';
+import styles from './launchesComparisonChart.scss';
+import { chartConfigs } from '../../chartConfigs';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  statistics$executions$total: {
+    id: 'FilterNameById.statistics$executions$total',
+    defaultMessage: 'Total',
+  },
+  statistics$executions$passed: {
+    id: 'FilterNameById.statistics$executions$passed',
+    defaultMessage: 'Passed',
+  },
+  statistics$executions$failed: {
+    id: 'FilterNameById.statistics$executions$failed',
+    defaultMessage: 'Failed',
+  },
+  statistics$executions$skipped: {
+    id: 'FilterNameById.statistics$executions$skipped',
+    defaultMessage: 'Skipped',
+  },
+  statistics$defects$product_bug: {
+    id: 'FilterNameById.statistics$defects$product_bug',
+    defaultMessage: 'Product bug',
+  },
+  statistics$defects$automation_bug: {
+    id: 'FilterNameById.statistics$defects$automation_bug',
+    defaultMessage: 'Automation bug',
+  },
+  statistics$defects$system_issue: {
+    id: 'FilterNameById.statistics$defects$system_issue',
+    defaultMessage: 'System issue',
+  },
+  statistics$defects$no_defect: {
+    id: 'FilterNameById.statistics$defects$no_defect',
+    defaultMessage: 'No defect',
+  },
+  statistics$defects$to_investigate: {
+    id: 'FilterNameById.statistics$defects$to_investigate',
+    defaultMessage: 'To investigate',
+  },
+  ofTestCases: {
+    id: 'Widgets.ofTestCases',
+    defaultMessage: 'of test cases',
+  },
+});
+
+@injectIntl
+@connect(
+  (state) => ({
+    project: activeProjectSelector(state),
+    getDefectLink: (params) => defectLinkSelector(state, params),
+    getStatisticsLink: (name) => statisticsLinkSelector(state, { statuses: [name] }),
+  }),
+  {
+    redirect,
+  },
+)
+export class LaunchesComparisonChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    redirect: PropTypes.func.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    height: PropTypes.number,
+    project: PropTypes.string.isRequired,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+    getDefectLink: PropTypes.func.isRequired,
+    getStatisticsLink: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeChart);
+  }
+
+  componentWillUnmount() {
+    this.node.removeEventListener('mousemove', this.getCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeChart);
+  }
+
+  onChartCreated = (chart, element) => {
+    this.chart = chart;
+    this.node = element;
+
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    const legendClassName = cx('legend');
```

Comment:
Create `setupLegend` method, witch will contain legend-related stuff.

Reply:
Moreover, legend should be common reusable, configurable component which can be used in others widgets. Please think about moving it outside.